src/Components/Authorization/src/AttributeAuthorizeDataCache.cs

```diff
@@ -29,13 +29,22 @@ private static IAuthorizeData[] ComputeAuthorizeDataForType(Type type)
         {
             // Allow Anonymous skips all authorization
             var allAttributes = type.GetCustomAttributes(inherit: true);
-            if (allAttributes.OfType<IAllowAnonymous>().Any())
+            List<IAuthorizeData> authorizeDatas = null;
+            for (var i = 0; i < allAttributes.Length; i++)
             {
-                return null;
+                if (allAttributes[i] is IAllowAnonymous)
+                {
+                    return null;
+                }
+
+                if (allAttributes[i] is IAuthorizeData authorizeData)
+                {
+                    authorizeDatas ??= new();
```

Comment:
(nit) would it be better to keep the existing behavior and not allocate this List if there is an `IAllowAnonymous`? (basically that means looping over the attributes twice.)

Reply:
At least with ASP.NET Core, having both sets of these attributes produce some side-effects (have the framework authenticate, but not authorize), but that work with Blazor. It would be fairly weird to have both of them applied. @javiercn in case you feel otherwise.

Reply:
> It would be fairly weird to have both of them applied

In that case, should the `new()` actually be `new(allAttributes.Length)`?  Or do we expect the type to have other attributes as well beyond those implementing these two interfaces?